examples/appSpec/zomes/three/capabilities/main/code/index.ts

```diff
@@ -38,6 +41,54 @@ export function test_get(encoded_allocation: u32): u32 {
 }
 
 
+/*----------  test_decode_params boilerplate  ----------*/
+
+
+var valueString: string = "";
+
+class ParameterHandler extends Handler {
+  onString(value: string): void {
+    if(this.currentKey == "keyString") {
+        valueString = value;
+    }
+  }
+}
+
+export function test_decode_params(encoded_allocation: u32): u32 {
+   let jsonString: string = deserialize(encoded_allocation);   
+
+    let handler = new ParameterHandler()
+    parseString<ParameterHandler>(jsonString, handler)
+
+   return serialize(valueString);
+
+}
+
+/*----------  test_decode_multiple_params boilerplate  ----------*/
+
+var string1: string = "";
+var string2: string = "";
+
+class MultiParameterHandler extends Handler {
+  onString(value: string): void {
+    if(this.currentKey == "firstString") {
```

Comment:
Could currentKey be passed in as a param to the function itself, rather than being an implicit property on the class?

Reply:
Yes I like your idea better. I think I will combine currentKey and keyStack to a single parameter that is passed to each of the value callbacks.